doc: Update Configuration and Development Overview (#7705)
projectacrn/acrn-hypervisor: doc/getting-started/overview_dev.rst

ACRN offers three types of VMs:

* **Pre-launched User VMs**: These VMs run independently of other VMs and own

>   dedicated hardware resources, such as a CPU core, memory, and I/O devices.

Reviewer: ```suggestion
  dedicated hardware resources, such as a CPU cores, memory, and I/O devices.
```

Reviewer 2: That's mixing singular and plural, isn't it? "a ... cores"

  Other VMs, including the Service VM, may not even be aware of a pre-launched
  VM's existence. The configuration of pre-launched VMs is static and must be
  defined at build time. They are well-suited for safety-critical applications
  and where very strict isolation, including from the Service VM, is desirable.

* **Service VM**: A special VM, required for scenarios that have post-launched

>   User VMs. The Service VM can access hardware resources directly by running

Reviewer: Can you check the indenting of the rest of this paragraph.  Something looks wonky here.

stringent real-time requirements, start by testing the application on a
post-launched VM before considering a pre-launched VM.

Scenario Types
---------------

ACRN categorizes scenarios into :ref:`three types <usage-scenarios>`:

* **Shared scenario:** This scenario represents a traditional computing, memory,
  and device resource sharing model among VMs. It has post-launched User VMs and
  the required Service VM. There are no pre-launched VMs in this scenario.

* **Partitioned scenario:** This scenario has pre-launched User VMs only. It
  demonstrates VM partitioning: the User VMs are independent and isolated, and

>   they do not share resources. There is no need for the Service VM or Device

Reviewer: It's worth adding that storage devices are not shared either, so each pre-launched VM requires its own boot device.

Inspector tool, because the tool records the current BIOS settings in the board
configuration file.

> ACRN requires the virtualization settings listed in :ref:`gsg-board-setup` of

Reviewer: There are more than just virtualization settings in the BIOS that are needed, and I think we should mention BIOS here.
```suggestion
ACRN requires the BIOS settings listed in :ref:`gsg-board-setup` of
```

|icon_host| Step 4: Generate a Scenario Configuration File and Launch Scripts
*****************************************************************************

> The :ref:`acrn_configurator_tool`, found in the ACRN hypervisor source code,

Reviewer: I believe we'll actually have a Debian package for the 3.0 release that will be used to install the configurator.
```suggestion
The :ref:`acrn_configurator_tool`
```

> enables you to configure your scenario settings via a graphical user interface

Reviewer: Less passive would be:
```suggestion
lets you configure your scenario settings via a graphical user interface
```

(GUI) on your development computer.

> The tool guides you to import the board configuration file that you generated in

Reviewer: Less passive would be:
```suggestion
The tool imports the board configuration file that you generated in
```
